crates/rethnet_rpc_server/src/lib.rs

```diff
@@ -62,9 +64,27 @@ pub enum MethodInvocation {
 
 type RethnetStateType = Arc<RwLock<dyn SyncState<StateError>>>;
 
+struct Filter {
+    // TODO: later, when adding in the rest of the filter methods, consider removing this `type`
+    // field entirely.  i suspect it's probably redundant as compared to the type implied by the
+    // variants in `events`, but that suspicion will be confirmed or denied by the addition of
+    // those other filter methods.
+    // r#type: SubscriptionType,
+    _criteria: Option<FilterOptions>,
```

Comment:
Sorry, I didn't address this before request a re-review because I thought you were just making a suggestion for future work (the next filter Issue), since the RPC methods in this Issue/PR don't actually use the filter criteria.  But I went ahead and drafted what you're suggesting here in this PR, in [this commit](https://github.com/NomicFoundation/hardhat/pull/4197/commits/090f6bf3aeebc4789bb494ecf738c098d5e7e38d); let me know how it lines up with what you were envisioning.